orangecontrib/educational/widgets/owkmeans.py

```diff
@@ -135,85 +137,71 @@ class OWKmeans(OWWidget):
               '#492970', '#f28f43', '#77a1e5', '#c42525', '#a6c96a']
 
     def __init__(self):
         super().__init__()
 
         # options box
-        self.optionsBox = gui.widgetBox(self.controlArea)
-        self.cbx = gui.comboBox(self.optionsBox, self, 'attr_x',
-                                label='X:',
-                                orientation=Qt.Horizontal,
-                                callback=self.restart,
-                                sendSelectedValue=True)
-        self.cbx.setSizePolicy(QSizePolicy(QSizePolicy.MinimumExpanding, QSizePolicy.Fixed))
-        self.cby = gui.comboBox(self.optionsBox, self, 'attr_y',
-                                label='Y:',
-                                orientation='horizontal',
-                                callback=self.restart,
-                                sendSelectedValue=True)
-        self.cby.setSizePolicy(QSizePolicy(QSizePolicy.MinimumExpanding, QSizePolicy.Fixed))
+        self.optionsBox = gui.widgetBox(self.controlArea, "Data")
+        opts = dict(
+            widget=self.optionsBox, master=self, orientation=Qt.Horizontal,
+            callback=self.restart, sendSelectedValue=True)
+        policy = QSizePolicy(QSizePolicy.MinimumExpanding, QSizePolicy.Fixed)
+        self.cbx = gui.comboBox(value='attr_x', label='X: ', **opts)
+        self.cbx.setSizePolicy(policy)
+        self.cby = gui.comboBox(value='attr_y', label='Y: ', **opts)
+        self.cby.setSizePolicy(policy)
 
         self.centroidsBox = gui.widgetBox(self.controlArea, "Centroids")
-        self.centroidNumbersSpinner = gui.spin(self.centroidsBox,
-                                               self,
-                                               'numberOfClusters',
-                                               minv=1,
-                                               maxv=10,
-                                               step=1,
-                                               label='Number of centroids:',
-                                               callback=self.number_of_clusters_change)
-        self.centroidNumbersSpinner.setSizePolicy(QSizePolicy(QSizePolicy.MinimumExpanding, QSizePolicy.Fixed))
-        self.restartButton = gui.button(self.centroidsBox, self, self.button_labels["random_centroids"],
-                                        callback=self.restart)
-        self.linesCheckbox = gui.checkBox(self.centroidsBox,
-                                          self,
-                                          'lines_to_centroids',
-                                          'Show membership lines',
-                                          callback=self.complete_replot)
+        self.centroidNumbersSpinner = gui.spin(
+            self.centroidsBox, self, 'numberOfClusters',
+            minv=2, maxv=10, step=1, label='Number of centroids:',
+            alignment=Qt.AlignRight, callback=self.number_of_clusters_change,
+            sizePolicy=policy)
```

Comment:
Subjective, again, but I prefer not to list arguments one below the other if it doesn't contribute to clarity.